### ISSUE TITLE
Mosdepth fails with could not import: pcre_free_study

### ISSUE THREAD

**Issue by johanneskoester** [bioconda/bioconda-recipes#8548]
Running `mosdepth -h` after installation via Bioconda fails with `could not import: pcre_free_study`. Strangely, the environment contains the most recent `pcre 8.41-1` package that was also used to build this mosdepth version... Also, prefixing the call with the appropriate LD_LIBRARY_PATH (which should not be necessary), still leads to the same result.

Any ideas?

cc: @brentp

**Comment by brentp** [comment]
not sure. maybe it's finding an old pcre somehow even though the newer one is installed?

```
locate libpcre.so | xargs -I{}  grep -L pcre_free_study {}
```
will show the libs were that symbol is not found and:
```
locate libpcre.so | xargs -I{}  grep -l pcre_free_study {}
```
will show where it is found.

**Comment by johanneskoester** [comment]
Yes, the system definitely has an old libpcre. The one delivered with conda as a dependency is fine though. So, my guess is that the build script does not set the search paths correctly, and accidentally picks libpcre from the system (i.e. the docker container). 

I think the build.sh should use `--cincludes=$PREFIX/include --clibdir="$PREFIX/lib` when invoking the nim compiler, shouldn't it?

cc @chapmanb

**Comment by chapmanb** [comment]
Johannes -- that sounds reasonable to me although I'll defer to Brent on this as I am far from an expert on building nim code and mostly just copy and paste from his scripts.

**Comment by brentp** [comment]
I am also unsure on this. I have a note in the mosdepth readme about this same error, but it just requires building a newer pcre. @johanneskoester your solution seems sane to me.

**Comment by brentp** [comment]
is there a way we can test for this in the build?
